src-worker/algorithms/group.ts

```diff
@@ -13,9 +13,7 @@ export class AlgorithmGroup {
     this.algorithmMap.set('post', new Post(config, proxyPool))
   }
 
-  getByType (algorithmType: AlgorithmType) : Algorithm | null {
-    const alg = this.algorithmMap.get(algorithmType)
-    if (alg === undefined) return null
```

Comment:
this is for backward compatibility. map may not contain newer algorithm and `this.algorithmMap.get(algorithmType)` returns `Algorithm | undefined` . So there was null check added. Its at least must be changed to return type `Algorithm | undefined`

Reply:
what if we set method to `get` if it's not provided?